packages/components/src/components/base/models/QueryInfo.ts

```diff
@@ -122,6 +122,7 @@ export class QueryInfo extends Record({
             Object.assign({}, queryInfoJson, {
                 columns,
                 schemaQuery,
+                views: Map<string, ViewInfo>(), // need views to be a Map to avoid 'get is not defined' errors
```

Comment:
nit: I'm curious why you were hitting an error. The default value is set `views: Map<string, ViewInfo>(),` above to a `Map`. Do you have a repro of this scenario?

It also seems odd to need to set this in the `fromJSON` method rather than `create`.

Reply:
It's because the JSON object captured from the output of `queryDetails` has a `views` key so the default value doesn't come into play, but `views` is just an array and that doesn't get converted to an immutable anything (we do a lot of doctoring of this before creating the actual map of views in the non-test case).  The error occurred in my tests when trying to use the `queryModel.allColumns` method.